std/traits.d

```diff
@@ -2525,11 +2525,17 @@ Returns:
 template isInnerClass(T)
 if (is(T == class))
 {
-    import std.meta : staticIndexOf;
-
     static if (is(typeof(T.outer)))
-        enum isInnerClass = __traits(isSame, typeof(T.outer), __traits(parent, T))
-                         && (staticIndexOf!(__traits(allMembers, T), "outer") == -1);
+    {
+        template hasOuterMember(T...)
+        {
+            static if (T.length == 0)
+                enum hasOuterMember = false;
+            else
+                enum hasOuterMember = T[0] == "outer" || hasOuterMember!(T[1 .. $]);
```

Comment:
Linear recursion template.

Reply:
It replaces a linear recursion template in a very deep import stack, so it's better than what's there.
I gotta hit the sack. You can log-N-ify it if you like....

Reply:
I've been thinking on it, and I don't think it's possible to improve on linear recursion :/
D needs C++'s `...` expression. That would eliminate 90% of recursive templates.